src/Discord.Net.WebSocket/Entities/Interaction/SocketInteractionDataOption.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Model = Discord.API.ApplicationCommandInteractionDataOption;
+
+namespace Discord.WebSocket
+{
+    /// <summary>
+    ///     Represents a Websocket-based <see cref="IApplicationCommandInteractionDataOption"/> recieved by the gateway
+    /// </summary>
+    public class SocketInteractionDataOption : IApplicationCommandInteractionDataOption
+    {
+        /// <inheritdoc/>
+        public string Name { get; private set; }
+
+        /// <inheritdoc/>
+        public object Value { get; private set; }
+
+        /// <summary>
+        ///      The sub command options recieved for this sub command group.
+        /// </summary>
+        public IReadOnlyCollection<SocketInteractionDataOption> Options { get; private set; }
+
+        private DiscordSocketClient discord;
+        private ulong guild;
+
+        internal SocketInteractionDataOption() { }
+        internal SocketInteractionDataOption(Model model, DiscordSocketClient discord, ulong guild)
+        {
+            this.Name = Name;
```

Comment:
This line should be `this.Name = model.Name;`

Reply:
I can confirm this fixes the problem I posted about

Reply:
Whoops yeah that might be an issue, I'l make a comit tomorrow regarding this and some other changes.